var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -50,10 +50,10 @@ class Legion(CMakePackage, ROCmPackage):
     depends_on("ucx", when="network=ucx")
     depends_on("ucx", when="conduit=ucx")
     depends_on("mpi", when="conduit=mpi")
-    depends_on("cuda@10.0:11.9", when="+cuda_unsupported_compiler @:23.03.0")
-    depends_on("cuda@10.0:11.9", when="+cuda @:23.03.0")
-    depends_on("cuda@10.0:12.2", when="+cuda_unsupported_compiler @23.06.0:")
-    depends_on("cuda@10.0:12.2", when="+cuda @23.06.0:")
+    depends_on("cuda@10.0:11.9", when="+cuda_unsupported_compiler @21.03.0:23.03.0")
+    depends_on("cuda@10.0:11.9", when="+cuda @21.03.0:23.03.0")
+    depends_on("cuda@10.0:12.2", when="+cuda_unsupported_compiler @23.06.0:,cr:")
+    depends_on("cuda@10.0:12.2", when="+cuda @23.06.0:,cr:")
```

Comment:
Doesn't this impose the dependency on _all_ numbered versions, since they're greater than `cr`?  Of course, this is a strictly weaker dependency than the one applied to all versions &le;`23.03.0`, so that doesn't actually change anything, but in that case we could just say
```suggestion
    depends_on("cuda@10.0:12.2", when="+cuda_unsupported_compiler")
    depends_on("cuda@10.0:12.2", when="+cuda")
```
and leave the versioning to the stronger dependency above.

Reply:
Yes, that simplifies it.